scripts/update-references-in-vsce.js

```diff
@@ -0,0 +1,164 @@
+function updatePackageJson() {
+  // Function to update dependencies in package.json
+  function updateDependencies(filePath) {
+    // Read package.json file
+    fs.readFile(filePath, { encoding: 'utf8' }, (err, data) => {
+      if (err) {
+        console.error(`Error reading file ${filePath}: ${err}`);
+        return;
+      }
+
+      let updated = false;
+      const json = JSON.parse(data);
+
+      // Check and update the dependencies according to sourceMap
+      for (const oldDep in referencesMap) {
+        const newDep = referencesMap[oldDep];
+        // Update dependencies 
+        if (json?.dependencies?.[oldDep]) {
+          json.dependencies[newDep] = json.dependencies[oldDep].startsWith('^') ? json.dependencies[oldDep] : '^' + json.dependencies[oldDep];
+          delete json.dependencies[oldDep];
+          updated = true;
+        }
+        // Update packaging dependencies
+        const packagingDeps = json?.packaging?.packageUpdates?.dependencies;
+        if (packagingDeps?.[oldDep]) {
+          packagingDeps[newDep] = packagingDeps[oldDep].startsWith('^') ? packagingDeps[oldDep] : '^' + packagingDeps[oldDep];
+          delete packagingDeps[oldDep];
+          updated = true;
+        }
+      }
+
+      // If updated, write the modified package.json back to disk
+      if (updated) {
+        fs.writeFile(filePath, JSON.stringify(json, null, 2), (err) => {
+          if (err) {
+            console.error(`Error writing file ${filePath}: ${err}`);
+          } else {
+            console.log(`Updated dependencies in ${filePath}`);
+          }
+        });
+      }
+    });
+  }
+
+  // Start traversing from the packages directory
+  traverseDirectories(packagesDir, 'package.json', updateDependencies);
+}
+
+function updateImports() {
+  const dirs = ['src', 'test'];
+  function replaceTextInFile(filePath) {
+    const data = fs.readFileSync(filePath, 'utf8');
+    const result = updateReferenceNames(data);
+    fs.writeFileSync(filePath, result, 'utf8');
+  }
+  function traverseDirectory(directory) {
+    fs.readdirSync(directory).forEach((file) => {
+      const fullPath = path.join(directory, file);
+      if (fs.lstatSync(fullPath).isDirectory()) {
+        traverseDirectory(fullPath);
+      } else if (path.extname(fullPath) === '.ts') {
+        replaceTextInFile(fullPath);
+      }
+    });
+  }
+
+  function traversePackages(directory) {
+    fs.readdir(directory, { withFileTypes: true }, (err, files) => {
+      if (err) {
+        console.error(`Error reading directory ${directory}: ${err}`);
+        return;
+      }
+
+      files.forEach(file => {
+        if (file.isDirectory()) {
+          dirs.forEach((dir) => {
+            const tsDirPath = path.join(directory, file.name, dir);
+            if (fs.existsSync(tsDirPath)) {
+              traverseDirectory(tsDirPath)
+            }
+          });
+        }
+      });
+    });
+  }
+  traversePackages(packagesDir);
```

Comment:
I found something interesting - if you replace Line 146 with `traverseDirectory(packagesDir)`, you will get the same result except for a difference in one file - `packages/salesforcedx-sobjects-faux-generator/scripts/regenerateMinSObjects.ts`.

When you use `traversePackages()`, the script uses @salesforce/core.
<img width="400" alt="Screenshot 2024-05-09 at 11 09 41 AM" src="https://github.com/forcedotcom/bundle-publish-scripts/assets/139700604/355cc1a8-f637-439c-9279-89ad4ff45c48">

When you use `traverseDirectory()`, the script replaces it with @salesforce/core-bundle.
<img width="450" alt="Screenshot 2024-05-09 at 11 10 17 AM" src="https://github.com/forcedotcom/bundle-publish-scripts/assets/139700604/da200184-4a4e-4188-92f2-2212a749dde4">

Can I assume you want to use @salesforce/core-bundle here?  If yes, the `traversePackages()` function is unused and can be removed.